src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -2544,7 +2555,22 @@ void VM_Version::crac_restore() {
   // Workaround JDK-8311164: CPU_HT is set randomly on hybrid CPUs like Alder Lake.
   features_missing &= ~CPU_HT;
 
-  if (features_missing || glibc_features_missing) {
+  if (CPUFeatures) {
+    if (nonlibc_str_equals(CPUFeatures, "ignore")) {
+      ignore_glibc_not_using = true;
+    } else {
+      static const char part1[] = "Unsupported -XX:CPUFeatures=";
+      tty->print_raw(part1, sizeof(part1) - 1);
+      for (const char *s = CPUFeatures; *s; ++s)
+        tty->put(*s);
+      static const char part2[] = ", only -XX:CPUFeatures=ignore is supported during -XX:CRaCRestoreFrom";
+      tty->print_raw(part2, sizeof(part2) - 1);
+      tty->cr();
+      vm_exit_during_initialization();
```

Comment:
Will this work when you set `CPUFeatures=generic` before checkpoint and don't set anything during restore? The option will keep its value and this might trigger the error.
If you don't want to add some extra tests you might piggyback in `VMOptionsTest`.

Reply:
Agree. And we are unable to handle e.g. `generic` specified for restore, although it's documented. 

Having dedicated IgnoreCPUFeatures would simplify the implementation a lot, and we'd also could make it Experimental, as it should be.

Reply:
There are currently no tests. Implemented the `-XX:+IgnoreCPUFeatures` option.